src/System.Linq/src/System/Linq/Concat.cs

```diff
@@ -11,19 +11,37 @@ public static partial class Enumerable
     {
         public static IEnumerable<TSource> Concat<TSource>(this IEnumerable<TSource> first, IEnumerable<TSource> second)
         {
-            if (first == null)
+            if (first is null)
             {
                 ThrowHelper.ThrowArgumentNullException(ExceptionArgument.first);
             }
 
-            if (second == null)
+            if (second is null)
             {
                 ThrowHelper.ThrowArgumentNullException(ExceptionArgument.second);
             }
 
-            return first is ConcatIterator<TSource> firstConcat
-                ? firstConcat.Concat(second)
-                : new Concat2Iterator<TSource>(first, second);
+            return ConcatIterator(first, second);
+
+            static IEnumerable<TSource> ConcatIterator (IEnumerable<TSource> first, IEnumerable<TSource> second)
```

Comment:
Style: There shouldn't be a space between the method name and parameters list.

```suggestion
            static IEnumerable<TSource> ConcatIterator(IEnumerable<TSource> first, IEnumerable<TSource> second)
```